docs/source/opener_parameters_api.md

```diff
@@ -0,0 +1,220 @@
+parameter, and other parameters whose values also follow such a format, are
+interpreted by xcube as a *restriction*, much like a bounding box or time
+range. That is:
+
+ - By default (if the parameter is omitted or if a `None` value is supplied
+   for it), *all* the possible member values MUST be included in the list. In
+   the case of `variables`, this will result in a dataset containing all the
+   available variables.
+ - If a list containing *some* of the possible members is given, a dataset
+   corresponding to those members only MUST be returned. In the case of
+   `variables`, this will result in a dataset containing only the requested
+   variables.
+ - As a special case of the above, if an empty list is supplied, a dataset
+   containing *no data* MUST be returned -- but with the requested spatial and
+   temporal dimensions.
+
+## Date, time, and duration specifications
+
+In the common parameter `time_range`, times can be specified using the
+standard JSON Schema formats `date-time` or `date`. Any additional time or
+date parameters supported by an xcube opener dataset SHOULD also use these
+formats, unless there is some good reason to prefer a different format.
+
+The formats are described in the [JSON Schema Validation 2019
+draft](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7.3.1),
+which in turn adopts definitions from [RFC 3339 Section
+5.6](https://tools.ietf.org/html/rfc3339#section-5.6). The JSON Schema
+`date-time` format corresponds to RFC 3339's `date-time` production, and JSON
+Schema's `date` format to RFC 3339's `full-date` production. These formats are
+subsets of the widely adopted [ISO
+8601](https://en.wikipedia.org/wiki/ISO_8601) format.
+
+The `date` format corresponds to the pattern `YYYY-MM-DD` (four-digit year –
+month – day), for example `1995-08-20`. The `date-time` format consists of a
+date (in the `date` format), a time (in `HH:MM:SS` format), and timezone (`Z`
+for UTC, or `+HH:MM` or `-HH:MM` format).  The date and time are separated by
+the letter `T`. Examples of `date-time` format include `1961-03-23T12:22:45Z`
+and `2018-04-01T21:12:00+08:00`. (Fractions of a second MAY also be included,
+but are unlikely to be relevant for xcube openers.)
+
+## Time limits: an extension to the JSON Schema
+
+JSON Schema itself does not offer a way to impose time limits on a string
+schema with the `date` or `date-time` format. This is a problem for xcube
+generator UI creation, since it might be reasonably expected that a UI will
+show and enforce such limits. The xcube opener API therefore defines an
+unofficial extension to the JSON string schema: a `JsonStringSchema` object
+(as returned as part of a `JsonSchema` by a call to
+`get_open_data_params_schema`) MAY, if it has a `format` property with a value
+of `date` or `date-time`, also have one or both of the properties
+`min_datetime` and `max_datetime`. These properties must also conform to the
+`date` or `date-time` format.
```

Comment:
```suggestion
`date` or `date-time` format.
xcube provides a dedicated `JsonDatetimeSchema` for this purpose. Internally, it extends
the `JsonStringSchema` by adding the required properties to the JSON string schema.
```

@pont-us Please add new class `JsonDatetimeSchema` and test. This will make code more readable!